indico_jacow/models/affiliations.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of the JACoW plugin.
+# Copyright (C) 2021 - 2024 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+from sqlalchemy.ext.declarative import declared_attr
+
+from indico.core.db.sqlalchemy import db
+from indico.core.db.sqlalchemy.util.models import auto_table_args
+from indico.util.decorators import strict_classproperty
+
+
+class MultipleAffiliationsBase(db.Model):
+    """Base class for multiple affiliations associations."""
+
+    __abstract__ = True
+    #: The name of the backref on the `Affiliation`
+    affiliations_backref_name = None
+    #: The name of the foreign key column on the person link table
+    person_link_fk = None
+    #: The name of the person link class
+    person_link_cls = None
+
+    @strict_classproperty
+    @classmethod
+    def __auto_table_args(cls):
+        return (db.Index(None, 'person_link_id', 'display_order', unique=True),
+                {'schema': 'plugin_jacow'})
```

Comment:
I think because `person_link_id` is an FK to a different model in the inheritance I get this error if I try to set the `__table_args__` directly:

```
sqlalchemy.exc.ArgumentError: Index 'ix_uq_abstract_affiliations_person_link_id_display_order' is against table 'abstract_affiliations', and cannot be associated with table 'contribution_affiliations'.
```

Reply:
Yes, you still need to use the `declared_attr`. But you should be able to just return your stuff in there.